controller/proxy/proxy.ml

```diff
@@ -1,57 +0,0 @@
-type credentials =
-  { user: string
-  ; password: string
-  }
-
-type t =
-  { credentials: credentials option
-  ; host: string
-  ; port: int
-  }
-
-let validate str =
-  let uri = Uri.of_string str in
-  if Uri.path uri = ""
-    && Uri.query uri = []
-    && Uri.fragment uri = None
-  then
-    match Uri.scheme uri, Uri.host uri, Uri.port uri with
-    | Some "http", Some host, Some port ->
-      Some
-        { credentials =
-          (match Uri.user uri, Uri.password uri with
-          | Some user, Some password -> Some { user; password }
-          | _ -> None)
-        ; host
-        ; port
-        }
-    | _ -> None
-  else
-    None
-
-let to_string ~hide_password t =
```

Comment:
The rationale behind having to specify `~hide_password` was to enforce the caller to say if he wanted to have the password visible explicitly. Now, I am wondering if `to_uri >> to_string` could be use by inadvertance instead of `pp`.

Reply:
Yeah, that seems like a fair concern.

What do you think about e6014fb?